Core/Src/main.c

```diff
@@ -87,14 +91,39 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  //default pulse 3200, 5% of 64000
+  setTimerPulseValue(3200);
+
+  // Start PWM signal on channel 1
+  if (HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1) != HAL_OK)
+  {
+	  // PWM Generation Error
+	  Error_Handler();
+  }
+
+  //set CS pin high default
+  HAL_GPIO_WritePin(CS_GPIO_Port, CS_Pin, GPIO_PIN_SET);
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  //set CS pin low to signal start of spi comms
+	  HAL_GPIO_WritePin(CS_GPIO_Port, CS_Pin, GPIO_PIN_RESET);
+	  HAL_SPI_Transmit(&hspi1, (uint8_t *)&START, 1, 100);
+	  HAL_SPI_TransmitReceive(&hspi1, (uint8_t *)&CONFIG, spi_buf, 2, 100);
```

Comment:
recommend you use just one `HAL_SPI_TransmitReceive` function call. implement with the array recommendations above.

additionally, arrays don't need crazy casting. `CONFIG` by default is a pointer to that array.

also, recommend use sizeof operator instead of magic number `2`

Reply:
Addressed :)